core/domain/exp_jobs_one_off.py

```diff
@@ -535,3 +536,41 @@ def reduce(key, values):
         # Combine all values from multiple lists into a single list
         # for that error type.
         yield (key, list(set().union(*final_values)))
+
+
+class ImageDataMigrationJob(jobs.BaseMapReduceOneOffJobManager):
+    """ One-off job for migrating the images in the exploration
+        from the GAE to GCS
+    """
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [file_models.FileSnapshotContentModel]
+
+    @staticmethod
+    def map(file_snapshot_content_model):
+        instance_id = file_snapshot_content_model.get_unversioned_instance_id()
+        filetype = instance_id[instance_id.rfind(".") + 1:]
+
+        if filetype in ALLOWED_IMAGE_EXTENSIONS:
+            filename = instance_id[instance_id.rfind("/") + 1:]
```

Comment:
This is incredibly messy. It also seems to make assumptions about the production database that need to be verified.

First, define your expected filename format using a regex. If the regex does not match the filename, then yield an error message with the problematic filename so that this can be caught and investigated when the job is run on the test server. See other jobs in the codebase for examples of how to do this. @bansalnitish can also help you if you are stuck, he has done this recently.

Second, when defining the regex, use capture groups to separate the different parts of the filename. That should be much simpler and lead to more semantically-clear code than putting `instance_id[1:]` all over the place.

Reply:
How to use capture group to get different parts of the file names?. 
I tried searching it but couldn't get much information. Could you give me a gist of what exactly are you referring to?
Also, by the expected filename format you mean the extension of the file?

Reply:
I'd suggest reading up on "regex capture groups" -- a google search for that phrase has plenty of references.

By expected filename format I mean a regex that describes the entire filename, not just its extension.

Reply:
Done